commands/commands.go

```diff
@@ -334,6 +335,46 @@ func logPanic(loggedError error) string {
 	return full
 }
 
+func ipAddresses() []string {
+	ips := make([]string, 0, 1)
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		ips = append(ips, "Error getting network interface: "+err.Error())
+		return ips
+	}
+	for _, i := range ifaces {
+		if i.Flags&net.FlagUp == 0 {
+			continue // interface down
+		}
+		if i.Flags&net.FlagLoopback != 0 {
+			continue // loopback interface
+		}
+		addrs, _ := i.Addrs()
+		s := ""
+		if err != nil {
+			ips = append(ips, "Error getting IP address: "+err.Error())
+			continue
+		}
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+			if ip == nil || ip.IsLoopback() {
```

Comment:
Should this check go before the `switch` above? You could have something like `(*net.IPNet)(nil)`, which would panic on L362 after trying to dereference it.

Reply:
I am not sure if I understand you right, but this doesn't panic:
```
                        addr = nil
			var ip net.IP
			switch v := addr.(type) {
			case *net.IPNet:
				ip = v.IP
			case *net.IPAddr:
				ip = v.IP
			}
```

Reply:
😱 totally disregard my suggestion -- I misread your code as `if addr == nill || addr.IsLoopback()`, instead of `ip`. This is all good 👍